src/pip/_internal/commands/list.py

```diff
@@ -160,32 +170,38 @@ def run(self, options, args):
             packages = self.get_uptodate(packages, options)
 
         self.output_package_listing(packages, options)
+        return SUCCESS
 
     def get_outdated(self, packages, options):
+        # type: (List[Distribution], Values) -> List[Distribution]
         return [
             dist for dist in self.iter_packages_latest_infos(packages, options)
             if dist.latest_version > dist.parsed_version
         ]
 
     def get_uptodate(self, packages, options):
+        # type: (List[Distribution], Values) -> List[Distribution]
         return [
             dist for dist in self.iter_packages_latest_infos(packages, options)
             if dist.latest_version == dist.parsed_version
         ]
 
     def get_not_required(self, packages, options):
-        dep_keys = set()
+        # type: (List[Distribution], Values) -> List[Distribution]
+        dep_keys = set()  # type: Set[Distribution]
         for dist in packages:
             dep_keys.update(requirement.key for requirement in dist.requires())
-        return {pkg for pkg in packages if pkg.key not in dep_keys}
+        return [pkg for pkg in packages if pkg.key not in dep_keys]
 
     def iter_packages_latest_infos(self, packages, options):
+        # type: (List[Distribution], Values) -> Distribution
         with self._build_session(options) as session:
             finder = self._build_package_finder(options, session)
 
             for dist in packages:
                 typ = 'unknown'
-                all_candidates = finder.find_all_candidates(dist.key)
+                all_candidates = finder.find_all_candidates(
+                    dist.key)  # type: ignore
```

Comment:
why this `type: ignore`?

Reply:
This is w.r.t. https://github.com/pypa/pip/pull/8016#issuecomment-613238221 , I commented out that line to see if the mypy tests still fail in CI, and they indeed did, but they are still passing in my laptop

Reply:
I was able to get the test passing, so I have removed the `type: ignore`